Nos/Models/CoreData/Event+CoreDataClass.swift

```diff
@@ -342,6 +345,112 @@ public class Event: NosManagedObject, VerifiableEvent {
         return fetchRequest
     }
     
+    /// A fetch request for all the events that should be cleared out of the database by 
+    /// `DatabaseCleaner.cleanupEntities(...)`.
+    ///
+    /// It will save the events for the given `user`, as well as other important events matching various other
+    /// criteria.
+    /// - Parameter before: The date before which events will be considered for cleanup. 
+    /// - Parameter user: The Author record for the currently logged in user. Special treatment is given to their data.
+    @nonobjc public class func cleanupRequest(before date: Date, for user: Author) -> NSFetchRequest<Event> {
+        let oldStoryCutoff = Calendar.current.date(byAdding: .day, value: -2, to: .now) ?? .now
+        
+        let request = NSFetchRequest<Event>(entityName: "Event")
+        request.sortDescriptors = [NSSortDescriptor(keyPath: \Event.receivedAt, ascending: true)]
+        let oldEventClause = "(receivedAt < %@ OR receivedAt == nil) AND referencingEvents.@count = 0" 
+        let notOwnEventClause = "(author != %@)"
+        let readStoryClause = "(isRead = 1 AND receivedAt > %@)"
+        let userReportClause = "(kind == \(EventKind.report.rawValue) AND " +
+        "authorReferences.@count > 0 AND eventReferences.@count == 0)"
+        let clauses = "\(oldEventClause) AND" +
+        "\(notOwnEventClause) AND " +
+        "NOT \(readStoryClause) AND " +
+        "NOT \(userReportClause)"
+        request.predicate = NSPredicate(
+            format: clauses,
+            date as CVarArg,
+            user,
+            oldStoryCutoff as CVarArg
+        )
+        
+        return request
+    }
+    
+    /// This constructs a predicate for events that should be protected from deletion when we are purging the database.
+    /// - Parameter before: The date before which events will be considered for cleanup. 
+    /// - Parameter user: The Author record for the currently logged in user. Special treatment is given to their data.
+    @nonobjc public class func protectedFromCleanupPredicate(for user: Author) -> NSPredicate {
+        // NOTE: This code is pretty much the same as that of `protectedFromCleanupSubqueryPredicate` but I can't 
```

Comment:
One way would be to have one function, but prepend `$event.` in one case but not in the other, for example, with a `isSubquery` argument. I think it doesn't look good nor polished, but at least it would be just one function. Maybe @joshuatbrown can provide another idea here.

Reply:
Yeah, I think your suggestion is good. I don't have a better idea at this point, but that's a straightforward, simple way to avoid needing to update both. Seems better so these don't get out of sync. Something like:

```
func protectedFromCleanupPredicate(for user: Author, isSubquery: Bool) {
    guard...
    let eventOrEmpty = isSubquery ? "$event." : "" // or let prefix, maybe?
    let userEventsPredicate = NSPredicate(format: "\(eventOrEmpty)author.hexadecimalPublicKey = '\(userKey)'")
    ...
}
```

I needed a visual to see how it would -- that seems fine to me, I think, and probably better than two methods.

Reply:
Awesome. Great suggestions as usual, Josh!

Reply:
@martindsq this was your idea! You get the credit! 😀